src/Console/Output/CheckstyleOutputFormatter.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\EasyCodingStandard\Console\Output;
+
+use DOMDocument;
+use DOMElement;
+use RuntimeException;
+use Symplify\EasyCodingStandard\Console\ExitCode;
+use Symplify\EasyCodingStandard\Console\Style\EasyCodingStandardStyle;
+use Symplify\EasyCodingStandard\Contract\Console\Output\OutputFormatterInterface;
+use Symplify\EasyCodingStandard\ValueObject\Configuration;
+use Symplify\EasyCodingStandard\ValueObject\Error\ErrorAndDiffResult;
+
+/**
+ * @see \Symplify\EasyCodingStandard\Tests\Console\Output\JsonOutputFormatterTest
+ */
+final readonly class CheckstyleOutputFormatter implements OutputFormatterInterface
+{
+    /**
+     * @var string
+     */
+    public const NAME = 'checkstyle';
+
+    public function __construct(
+        private EasyCodingStandardStyle $easyCodingStandardStyle,
+        private ExitCodeResolver $exitCodeResolver
+    ) {
+    }
+
+    /**
+     * @return ExitCode::*
+     */
+    public function report(ErrorAndDiffResult $errorAndDiffResult, Configuration $configuration): int
+    {
+        $checkstyleContent = $this->createCheckstyleContent($errorAndDiffResult);
+        $this->easyCodingStandardStyle->writeln($checkstyleContent);
+
+        return $this->exitCodeResolver->resolve($errorAndDiffResult, $configuration);
+    }
+
+    public function getName(): string
+    {
+        return self::NAME;
+    }
+
+    /**
+     * @api
+     */
+    public function createCheckstyleContent(ErrorAndDiffResult $errorAndDiffResult): string
+    {
+        if (! \extension_loaded('dom')) {
+            throw new RuntimeException('Cannot generate report! `ext-dom` is not available!');
```

Comment:
Could you add composer suggest config on target-repository composer.json?

https://github.com/easy-coding-standard/easy-coding-standard/blob/main/build/target-repository/composer.json#L21

Reply:
Added.